internal/app/machined/pkg/controllers/kubespan/nftables.go

```diff
@@ -324,6 +363,130 @@ func matchIPSet(set *nftables.Set, mark, mask uint32, family nftables.TableFamil
 	}
 }
 
+func matchIPSetMSS(set *nftables.Set, mtu uint32, family nftables.TableFamily) []expr.Any {
+	var (
+		offset uint32 = 16
+		length uint32 = 4
+		ifmtu  uint16
+		mss    uint16
+	)
+
+	switch {
+	case mtu == 0:
+		ifmtu = 1500
+	case mtu < 576: // Minimum IPv4 mtu size
+		ifmtu = 576 - 40
+	default:
+		ifmtu = uint16(mtu)
+	}
+
+	mss = ifmtu - 40
+
+	if family == nftables.TableFamilyIPv6 {
+		offset = 24
+		length = 16
+		mss = ifmtu - 60
+	}
+
+	// ip daddr @kubespan_targets_ipv4 tcp flags & (syn|rst) == syn tcp option maxseg size > $MSS tcp option maxseg size set $MSS
+	return []expr.Any{
+		// Store protocol type to register 1
+		&expr.Meta{
+			Key:      expr.MetaKeyNFPROTO,
+			Register: 1,
+		},
+		// Match IP Family
+		&expr.Cmp{
+			Op:       expr.CmpOpEq,
+			Register: 1,
+			Data:     []byte{byte(family)},
+		},
+
+		// Store the destination IP address to register 1
+		&expr.Payload{
+			DestRegister: 1,
+			Base:         expr.PayloadBaseNetworkHeader,
+			Offset:       offset,
+			Len:          length,
+		},
+		// Match from target set
+		&expr.Lookup{
+			SourceRegister: 1,
+			SetName:        set.Name,
+			SetID:          set.ID,
+		},
+
+		// Load the current packet mark into register 1
+		&expr.Meta{
+			Key:      expr.MetaKeyL4PROTO,
+			Register: 1,
+		},
+		// Match TCP Family
+		&expr.Cmp{
+			Op:       expr.CmpOpEq,
+			Register: 1,
+			Data:     []byte{unix.IPPROTO_TCP},
+		},
+
+		// [ payload load 1b @ transport header + 13 => reg 1 ]
+		&expr.Payload{
+			DestRegister: 1,
+			Base:         expr.PayloadBaseTransportHeader,
+			Offset:       13,
+			Len:          1,
+		},
+		// [ bitwise reg 1 = ( reg 1 & 0x00000006 ) ^ 0x00000000 ]
+		&expr.Bitwise{
+			DestRegister:   1,
+			SourceRegister: 1,
+			Len:            1,
+			Mask:           []byte{0x02 | 0x04},
+			Xor:            []byte{0x00},
+		},
+		// [ cmp eq reg 1 0x00000002 ]
+		&expr.Cmp{
+			Op:       expr.CmpOpEq,
+			Register: 1,
+			Data:     []byte{0x02},
+		},
+
+		// [ exthdr load tcpopt 2b @ 2 + 2 => reg 1 ]
+		&expr.Exthdr{
+			DestRegister: 1,
+			Type:         2,
+			Offset:       2,
+			Len:          2,
+			Op:           expr.ExthdrOpTcpopt,
+		},
+		// [ cmp gte reg 1 MTU ]
+		&expr.Cmp{
+			Op:       expr.CmpOpGt,
+			Register: 1,
+			Data:     binaryutil.BigEndian.PutUint16(mss),
+		},
+		// [ immediate reg 1 MTU ]
+		&expr.Immediate{
+			Register: 1,
+			Data:     binaryutil.BigEndian.PutUint16(mss),
+		},
+		// [ exthdr write tcpopt reg 1 => 2b @ 2 + 2 ]
+		&expr.Exthdr{
+			SourceRegister: 1,
+			Type:           2,
+			Offset:         2,
+			Len:            2,
+			Op:             expr.ExthdrOpTcpopt,
+		},
+	}
+}
+
+func ifname(n string) []byte {
+	b := make([]byte, 16)
+	copy(b, []byte(n+"\x00"))
```

Comment:
```suggestion
	copy(b, []byte(n))
```

as `b` is all zeroes, there should be no reason to append `0`